internal/command/launch/launch.go

```diff
@@ -90,21 +92,16 @@ func run(ctx context.Context) (err error) {
 		workingDir = absDir
 	}
 
-	var importedConfig bool
-	appConfig := appconfig.NewConfig()
-
 	configFilePath := filepath.Join(workingDir, appconfig.DefaultConfigFileName)
-	if exists, _ := appconfig.ConfigFileExistsAtPath(configFilePath); exists {
-		cfg, err := appconfig.LoadConfig(configFilePath)
-		if err != nil {
-			return err
-		}
+	importedConfig, appConfig := false, appconfig.NewConfig()
 
+	// a fly toml already exists
+	if cfg != nil {
 		if cfg.AppName != "" {
 			fmt.Fprintln(io.Out, "An existing fly.toml file was found for app", cfg.AppName)
-			if deployExisting, err := shouldDeployExistingApp(ctx, cfg.AppName); err != nil {
+			if running, err := isAppRunning(ctx, cfg); err != nil {
```

Comment:
IMHO we should remove this shortcut to fast deploy completely. It is evil and unexpected DX, we already ask people if they want to deploy at the end of `fly launch` and otherwise advise to  do it later with `fly deploy`

Reply:
I agree; I like the idea of associating `fly launch` firmly with new-app setup magic, and `fly deploy` with deploying changes.

The fact that `fly launch` does this is, I believe, in response to the regular occurrence of users launching repeatedly when their first launch fails (which they can do because by default we'll give them new apps with generated names). It might not be the best possible solution though.

Reply:
I'm 100% with you @dangra  (I held the same opinion while fixing it)
But what's the consensus here? 

Should I rip it out? Or leave as is (@catflydotio 👁️ )

Reply:
The concern about creating apps on every `fly launch` is valid tho.

We have to do something about it, 🧠 dumping options next:

1. When offering to select a new name and no `--app` is passed, offer the current appConfig.AppName
2. but show a message saying the app exists, "blah blah... should continue?"
3. If YES, we won't recreate the app but must check its platformVersion matches `fly launch` expectation.
4. ... no idea, i think that is all...

Reply:
In order of preference:
1. `--generate-name` generate a random name if set
2. `--name` use this if set (IMO it should error if used in conjunction with `--generate-name`)
3. If interactive, show dialog to enter a new name, if fly.toml (aka appConfig) has a app name, use it as default value but let the user modify and confirm it
4. if non-interactive, fail as it does now.

Always check if app exists, and offer to proceed or stop there, and `--yes` to proceed no matter what

Reply:
(Whatever behaviour you settle on, it should just be clear to users what it is and what they need to do.)

Reply:
As it is right and once we remove the shortcut-unless-deployed, the scanner hooks are going to run no matter what. 

I wasn't sure we want to scan and run hooks on every `fly launch` retry but according to https://github.com/superfly/flyctl/pull/1730#issuecomment-1435264845 it is important not to skip them.

So, my proposal is to remove the shortcut, make it easy to reuse the app name from fly.toml, yet check if it exists and ask for confirmation to ~screw~ deploy into that existent app assuming all the consequences.

Reply:
This makes sense to me.

Reply:
@gwuah buying? do you think it is possible to implement this flow instead?

Reply:
yeah @dangra i should be able to pull this off. sounds good, but i'll give it some extra thought too.
(sorry for the late feedback, i've been immersing myself into networking @ fly)

Reply:
I've created a new pull request [here](https://github.com/superfly/flyctl/pull/1908).

For purposes of fixing the bug, we can merge this current pull request, and then rethink the user experience in https://github.com/superfly/flyctl/pull/1908

Reply:
Not sure, since this discussion thread, we have changed the flow so much that this PR is not a bugfix anymore